### PR TITLE
MRG+1: plot_trans:  add MRI fiducials

### PR DESCRIPTION
**Pull request by christianbrodbeck**
@Eric89GXL first component of https://github.com/mne-tools/mne-python/pull/3466

I made them smaller but same color as head fiducials to indicate the correspondence, do you think that works? Maybe slightly bigger?

The testing data has no fiducials file, if we would add one (should be very small) we could add a test...

fid

### CODE REVIEW

**Comment by Eric89GXL** [comment]
For testing purposes you can just make up a file, right? Use `_TempDir`, create dict, write with `write_dig`, etc.

**Comment by Eric89GXL** [comment]
I think the color is fine. But we should probably try to get in #3991, it already unifies some of these bits of code

**Comment by Eric89GXL** [comment]
Should be ready for rebase now

**Comment by agramfort** [comment]
if @Eric89GXL and Cis are happy I am happy.

**Comment by christianbrodbeck** [comment]
@Eric89GXL HPI-color is missing from defaults, is there no previous default?

**Comment by Eric89GXL** [comment]
I didn't transcribe all of them. It looks like it is `(0., 1., 0.)`.

**Comment by christianbrodbeck** [comment]
that does not seem ideal because it’s the same as the nasion color… do you think we can change it?

**Comment by Eric89GXL** [comment]
Yeah I think it's okay

**Comment by christianbrodbeck** [comment]
Yellow? Is it intentional that the head is now transparent even if not showing any surface inside the head?

screen shot 2017-02-20 at 12 35 30 pm

**Comment by Eric89GXL** [comment]
> Yellow?

We do use these for sources in source space. But maybe that's okay.

> Is it intentional that the head is now transparent even if not showing any surface inside the head?

No :(

**Comment by christianbrodbeck** [comment]
This is where head alpha comes from:

```
    alphas = (4 - np.arange(len(skull) + 1)) * (0.5 / 4.)
    head_alpha = alphas[0]
```

Should I just substitute 1 if len(skull) is 0?

**Comment by Eric89GXL** [comment]
I think the value of `0.5` should be `1.0` in the case that there is nothing inside the skull surfaces (no `src` and no `brain`).

**Comment by christianbrodbeck** [comment]
Do you think this works? Or Cyan? Magenta?

screen shot 2017-02-20 at 1 12 37 pm

screen shot 2017-02-20 at 1 03 44 pm

**Comment by Eric89GXL** [comment]
I find the doubled yellow a bit distracting. Cyan or magenta are fine by me

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -626,19 +662,29 @@ def plot_trans(info, trans='auto', subject=None, subjects_dir=None,
...
+              defaults['extra_color'], defaults['ecog_color']]
+    alphas = [0.8,
+              0.5,
+              0.25, 0.8]
+    scales = [defaults['eeg_scale'],
+              defaults['hpi_scale'],
+              defaults['extra_scale'], defaults['ecog_scale']]
+    for kind, loc in (('dig', car_loc), ('mri', fid_loc)):
+        if len(loc) > 0:
+            datas.extend(loc[np.newaxis, i, :] for i in range(3))
```
Comment (Eric89GXL) [inline mne/viz/_3d.py]: This is the same as `datas.exend(loc[:, np.newaxis])`

**Comment by christianbrodbeck** [comment]
screen shot 2017-02-20 at 2 59 32 pm
screen shot 2017-02-20 at 2 59 34 pm

**Comment by Eric89GXL** [comment]
LGTM, +1 for merge

@agramfort or @jaeilepp any opinion?

File: mne/viz/_3d.py [hunk 2]
```diff
@@ -320,8 +321,9 @@ def plot_trans(info, trans='auto', subject=None, subjects_dir=None,
         show EEG sensors in their digitized locations or projected onto the
         scalp, or a list of these options including ``[]`` (equivalent of
         False).
-    dig : bool
-        If True, plot the digitization points.
+    dig : bool | 'fiducials'
+        If True, plot the digitization points. "fiducials" to plot fiducial
+        points only.
```
Comment (agramfort) [inline mne/viz/_3d.py]: "fiducials" to plot fiducial  points only.
->
Use "mri_fiducials" to plot fiducial  points only.

Comment (Eric89GXL) [inline mne/viz/_3d.py]: I think it would be `Use "fiducials" to ...` since fiducials are separate from `mri_fiducials`

**Comment by agramfort** [comment]
arfff ok. +1 for MRG then

**Comment by Eric89GXL** [comment]
@jaeilepp want to look? If you don't have enough time then we have enough +1 for merge

File: mne/coreg.py [hunk 3]
```diff
@@ -19,6 +19,7 @@
 from numpy import dot
 
 from .io import read_fiducials, write_fiducials
+from .io.constants import FIFF
```
Comment (jaeilepp) [inline mne/coreg.py]: This needs to be removed.

**Comment by Eric89GXL** [comment]
I killed Travis because it's been waiting 3 hours for an OSX CI, but we should be okay.

Thanks @christianbrodbeck

**Pull request merged**